operator/src/main/java/org/projectnessie/operator/resource/Nessie.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.fabric8.kubernetes.api.model.Namespaced;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.Version;
+
+@Version(Nessie.VERSION)
+@Group(Nessie.GROUP)
+public class Nessie extends CustomResource<NessieSpec, NessieStatus> implements Namespaced {
+
+  public static final String GROUP = "nessie.projectnessie.org";
+  public static final String VERSION = "v1alpha1";
+  public static final String PLURAL = "nessies";
+  public static final String NAME = PLURAL + "." + GROUP;
```

Comment:
`nessies.nessie.projectnessie.org`?.. Do we need `nessie.` in the `GROUP`?

Reply:
I invoked operator-sdk with a non-empty group argument:

```
operator-sdk create api --plugins=quarkus --group nessie --version=v1alpha1 --kind=Nessie
```

The result was a group `nessie.projectnessie.org` under the domain `projectnessie.org`.

Indeed it's possible to place a CRD under a group that's just the domain name. I will see how I can amend this.

Reply:
Yes - let's keep `nessie` here. We have other projects (e.g. CEL) under `org.projectnessie` as well. Also conforms to the Maven group IDs now (we "moved" from `org.projectnessie` to `org.projectnessie.nessie` a while ago).

Reply:
Hmm OK, indeed I turned the project to "single-group" as per @dimas-b request in the meanwhile, meaning that the CRDs in it will have the same group, and that group is equal to the domain, `projectnessie.org`. So the Nessie CRD currently is `nessies.projectnessie.org`.

Note that we can have more than one CRD in a single-group project. E.g. we could have 2 CRDs:

* `nessies.projectnessie.org`
* `foos.projectnessie.org`

If we go back to multi-group, then we could have CRDs like:

* `nessies.nessie.projectnessie.org`
* `foos.nessie.projectnessie.org`
* `bars.other-group.projectnessie.org`
* `qix.other-group.projectnessie.org`

If that's OK, I'm going to go back to multi-group as per @snazy request.

Reply:
sgtm